fuel-vm/src/constraints.rs

```diff
@@ -99,13 +99,13 @@ impl CheckedMemRange {
 
     /// Create a new memory range, checks that the range fits into the constraint.
     fn new_inner(address: usize, size: usize, constraint: core::ops::Range<Word>) -> Result<Self, RuntimeError> {
-        if size == 0 {
-            return Ok(Self(address..address));
-        }
-
         let (end, of) = address.overflowing_add(size);
         let range = address..end;
-        if of || !constraint.contains(&((range.end - 1) as Word)) {
+
+        if of
+            || range.start != 0 && !constraint.contains(&((range.start - 1) as Word))
+            || range.end != 0 && !constraint.contains(&((range.end - 1) as Word))
```

Comment:
If I have a constraint of 1..5, but my range is 0..5, would this allow constraint check to be incorrectly skipped because the range.start == 0?